File: WeatherApp_Android-main/app/src/main/java/kth/kenanmark/weatherapp/model/Weather.kt
package kth.kenanmark.weatherapp.model

data class Weather(
    var temperatures: List<Parameter> = listOf<Parameter>(),
    var tccs: List<Parameter> = listOf<Parameter>(),
    var validTimes: List<String> = listOf<String>(),
    var approvedTime: String = "",
    var lat: Float = 0.0f,
    var long: Float = 0.0f)


data class WeatherResponse(
    var approvedTime: String,
    var timeSeries: List<TimeSeriesInfo>
)

data class TimeSeriesInfo(
    var validTime: String,
    var parameters: List<Parameter>
)

data class Parameter(
    var name: String,
    var values: List<Float>
)

File: WeatherApp_SwiftWeb-main/Sources/TokamakApp/Weather.swift
import Foundation
import JavaScriptKit

struct Weather: Codable{
    var temperatures = [Parameter]()
    var tccs = [Parameter]()
    var validTimes = [String]()
    var approvedTime = ""
    var lat: Float = 0.0
    var long: Float = 0.0
}

struct WeatherResponse: Codable{
    var approvedTime: String
    var timeSeries:[TimeSeriesInfo]
}

struct TimeSeriesInfo: Codable{
    var validTime:String
    var parameters:[Parameter]
}

struct Parameter: Codable{
    var name: String
    var values: [Float]
}

